OpenDreamRuntime/ServerVerbSystem.cs

```diff
@@ -130,35 +132,89 @@ private void OnPlayerStatusChanged(object? sender, SessionStatusEventArgs e) {
         RaiseNetworkEvent(new AllVerbsEvent(_verbs), e.Session);
     }
 
-    private void OnVerbExecuted(ExecuteVerbEvent msg, EntitySessionEventArgs args) {
-        var connection = _dreamManager.GetConnectionBySession(args.SenderSession);
-        var src = _dreamManager.GetFromClientReference(connection, msg.Src);
-        if (src == null || !_verbIdToProc.TryGetValue(msg.VerbId, out var verb) || !CanExecute(connection, src, verb))
-            return;
+    public void RunRepeatingVerbs() {
+        using (Profiler.BeginZone("Repeating Verbs", color: (uint)Color.OrangeRed.ToArgb())) {
+            foreach (var repeatingVerb in _repeatingVerbs) {
+                if (repeatingVerb.Value.Count == 0)
+                    return;
+                var client = repeatingVerb.Key;
+                var verbId = repeatingVerb.Value.Last();
+
+                var src = _dreamManager.GetFromClientReference(client, ClientObjectReference.Client);
+                if (src == null || !_verbIdToProc.TryGetValue(verbId, out var verb) || !CanExecute(client, src, verb))
+                    return;
+
+                using (Profiler.BeginZone("DM Execution", color: (uint)Color.LightPink.ToArgb()))
+                    DreamThread.Run($"Execute repeating {verb} by {client.Session!.Name}", async state => {
+                        await state.Call(verb, src, client.Mob);
+                        return DreamValue.Null;
+                    });
+            }
+        }
+    }
 
-        var argCount = verb.ArgumentTypes?.Count ?? 0;
-        if (msg.Arguments.Length != argCount) {
-            _sawmill.Error(
-                $"User \"{args.SenderSession.Name}\" gave {msg.Arguments.Length} argument(s) to the \"{verb.Name}\" verb which only has {argCount} argument(s)");
+    public void RemoveConnectionFromRepeatingVerbs(DreamConnection connection) {
+        _repeatingVerbs.Remove(connection);
+    }
+
+    private void OnRepeatVerbStart(RegisterRepeatVerbEvent msg, EntitySessionEventArgs args) {
+        if (!_verbIdToProc.ContainsKey(msg.VerbId))
             return;
+        var conn = _dreamManager.GetConnectionBySession(args.SenderSession);
+        if (_repeatingVerbs.TryGetValue(conn, out var verb)) {
+            if (!verb.Remove(msg.VerbId)) {
+                verb.Add(msg.VerbId);
+            }
```

Comment:
Why does this _stop_ executing the verb if it's already in the list?

Reply:
Whoops, meant for `Contains` here, thanks.